### PR TITLE
get quota for user only when needed

### PR DESCRIPTION
**Pull request by icewind1991**
always apply the wrapper but have the wrapper get the quota lazily only when an operation where quota is applied is performed

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by blizzz** [comment]
CI is unhappy

**Comment by icewind1991** [comment]
tests fixed

File: lib/private/Files/Storage/Wrapper/Quota.php [hunk 1]
```diff
@@ -128,6 +138,9 @@ public function free_space($path) {
 	 * @return int|false
 	 */
 	public function file_put_contents($path, $data) {
+		if (!$this->hasQuota()) {
+			return $this->storage->file_put_contents($path, $data);
```
Comment (blizzz) [inline lib/private/Files/Storage/Wrapper/Quota.php:142]: previously it went into checking whether enough space is left to write the file, now this is being skipped? Unless this is also computed somewhere up the hasQuota path.

Comment (icewind1991) [reply lib/private/Files/Storage/Wrapper/Quota.php:142]: previously the wrapper wouldn't be applied in the first place so the code path would have never been hit

**Review by blizzz — APPROVED** [review]

**Pull request merged**